nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/logging/repository/StandardLogRepository.java

```diff
@@ -82,14 +83,25 @@ public void addLogMessage(final LogLevel level, final String format, final Objec
         addLogMessage(level, formattedMessage, t);
     }
 
-    private void replaceThrowablesWithMessage(Object[] params) {
+    private void replaceThrowablesWithMessage(final Object[] params) {
         for (int i = 0; i < params.length; i++) {
-            if(params[i] instanceof Throwable) {
-                params[i] = ((Throwable) params[i]).getLocalizedMessage();
+            if (params[i] instanceof Throwable) {
+                params[i] = getCauses((Throwable) params[i]);
             }
         }
     }
 
+    private static String getCauses(final Throwable throwable) {
+        final StringBuilder builder = new StringBuilder();
```

Comment:
The code (and the Jira) suggests that the root cause exception should be at the top (`Causes` instead of `Caused by`). This would mean the order of the exception messages in the builder should be reversed.

Unfortunately however this might not even be the right place to handle the exceptions. This works if an unexpected exception is caught and subsequently reported by the framework. However direct calls to the `ComponentLog` methods from the processor don't produce the wanted result.